apps/meteor/app/apps/server/converters/messages.js

```diff
@@ -14,6 +14,59 @@ export class AppMessagesConverter {
 		return this.convertMessage(msg);
 	}
 
+	async convertMessageRaw(msgObj) {
+		if (!msgObj) {
+			return undefined;
+		}
+
+		const { rid, editedBy, attachments, u, ...message } = msgObj;
+
+		const getRoom = () => ({ id: rid });
+
+		const getEditor = () => (editedBy ? { id: editedBy._id, username: editedBy.username } : undefined);
+
+		const getAttachments = async () => this._convertAttachmentsToApp(attachments);
+
+		const getSender = () => {
+			if (!u || !u._id) {
+				return undefined;
+			}
+
+			const user = {
+				id: u._id,
+				username: u.username,
+				name: u.name,
+			};
+
+			return user;
+		};
+
+		const map = {
+			id: '_id',
+			threadId: 'tmid',
+			reactions: 'reactions',
+			parseUrls: 'parseUrls',
+			text: 'msg',
+			createdAt: 'ts',
+			updatedAt: '_updatedAt',
+			editedAt: 'editedAt',
+			emoji: 'emoji',
+			avatarUrl: 'avatar',
+			alias: 'alias',
+			file: 'file',
+			customFields: 'customFields',
+			groupable: 'groupable',
+			token: 'token',
+			blocks: 'blocks',
+			room: getRoom,
+			editor: getEditor,
+			attachments: getAttachments,
+			sender: getSender,
```

Comment:
IMO those fields do not need fancy conversions, we can just pass them through, i.e.:
```
roomId: 'rid',
editor: 'editedBy',
sender: 'u',
```

And then they require a new type on the apps-engine like `IMessageRaw`. This is important so it doesn't create the expectation on the app developer that the `room` field will have all the info because it is the same name and type of `IMessage.room`